src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -67,16 +72,104 @@ internal struct ParsedStat
             //internal long cguest_time;
         }
 
+        internal struct ParsedMapsModule
+        {
+            internal string FileName;
+            internal KeyValuePair<long, long> AddressRange;
+        }
+
         internal static string GetStatFilePathForProcess(int pid)
         {
             return RootPath + pid.ToString(CultureInfo.InvariantCulture) + StatFileName;
         }
 
+        internal static string GetMapsFilePathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + MapsFileName;
+        }
+
         internal static string GetTaskDirectoryPathForProcess(int pid)
         {
             return RootPath + pid.ToString(CultureInfo.InvariantCulture) + TaskDirectoryName;
         }
 
+        internal static IEnumerable<ParsedMapsModule> ParseMapsModules(int pid)
+        {
+            try
+            {
+                return ParseMapsModulesCore(File.ReadAllLines(GetMapsFilePathForProcess(pid)));
+            }
+            catch (FileNotFoundException) { }
+            catch (DirectoryNotFoundException) { }
+            catch (UnauthorizedAccessException) { }
+
+            return Array.Empty<ParsedMapsModule>();
+        }
+
+        private static IEnumerable<ParsedMapsModule> ParseMapsModulesCore(IEnumerable<string> lines)
+        {
+            Debug.Assert(lines != null);
+
+            // Parse each line from the maps file into a ParsedMapsModule result
+            foreach (string line in lines)
+            {
+                // Use a StringParser to avoid string.Split costs
+                var parser = new StringParser(line, separator: ' ', skipEmpty: true);
+
+                // Parse the address range 
+                KeyValuePair<long, long> addressRange =  
+                    parser.ParseRaw(delegate (string s, ref int start, ref int end)
```

Comment:
I don't have a strong opinion on it; personally I think it's a trivial difference and don't think we need to be strict about it, but I'm fine if others disagree and want to codify it one way or the other.  At that point though it should be built into codeformatter and the rule applied consistently across the repo.  Separate from that I'm going to leave this as-is.